tests/integration/schema/filter_test.go

```diff
@@ -287,3 +287,126 @@ var defaultBookArgsWithoutFilter = trimFields(
 	},
 	testFilterForOneToOneSchemaArgProps,
 )
+
+func TestSchemaFilterInputs_WithJSONField_Succeeds(t *testing.T) {
+	test := testUtils.TestCase{
+		Actions: []any{
+			testUtils.SchemaUpdate{
+				Schema: `
+					type Users {
+						custom: JSON
+					}
+				`,
+			},
+			testUtils.IntrospectionRequest{
+				Request: `
+					query {
+						__schema {
```

Comment:
suggestion: You should be able to query `__type` with a `name: "User"` filter - it will save the test some processing, and should make it much easier to debug, as only the requested type will be returned from the query (instead of every gql type in the system).

`TestSchemaSimple_WithJSONField_CreatesSchemaGivenType` does this in your other introspection test.